lib/msprime.c

```diff
@@ -2116,6 +2119,35 @@ msp_common_ancestor_event(msp_t *self, population_id_t population_id)
 }
 
 
+/* This calculates the rate given by Eq (2) in the notes
+ */
+static double
+msp_compute_lambda_Xi_dirac(unsigned int b, double psi, double c)
```

Comment:
I see. I would prefer option (1) then. Since I will be able to parse psi through model.params.dirac_coalescent.psi directly